File: typescript-javascript/algorithms/easy/simpleArraySum.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-04-24
 * Last modified: 2026-06-11
 * @description Simple Array Sum - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/simple-array-sum/problem}
 */
//---------- functions ----------
/**
 * Complete the 'simpleArraySum' function below.
 *
 * @param {number[]} ar The array to sum.
 * @returns {number} The function is expected to return an INTEGER.
 */
function simpleArraySum(ar: number[]): number 
{ return (ar.reduce((a, b) => a + b)); }

File: python/algorithms/easy/simpleArraySum.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-04-24
Last modified: 2026-06-13
Description: Simple Array Sum - HackerRank
Test description: https://www.hackerrank.com/challenges/solve-me-first/problem
"""
#---------- Imports ----------
import functools


#---------- functions ----------
def simpleArraySum(ar: list[int]) -> int:
    """
    Receive an array and sum all itens inside of it.

    Parameters:
    ar (list[int]): Array of numbers. 

    Returns:
    int: The sum of all values in array.
    """
    return functools.reduce(lambda x, y: x + y, ar)